ccaas_builder/cmd/detect/main.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var logger = log.New(os.Stderr, "", 0)
+
+func main() {
+	logger.Println("::Detect")
+
+	if err := run(); err != nil {
+		logger.Printf("::Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	logger.Printf("::Type detected as ccaas")
+
+}
+
+type ChaincodeMetadata struct {
+	Type string `json:"type"`
+}
+
+func run() error {
+	if len(os.Args) > 2 {
+		chaincodeMetaData := os.Args[2]
+		mdbytes, err := ioutil.ReadFile(filepath.Join(chaincodeMetaData, "metadata.json"))
+		if err != nil {
+			return err
```

Comment:
We typically add context to the error so the user knows what was going on instead of just returning the underlying Go error message. Typically `errors.WithMessage()` is used for this:
https://pkg.go.dev/github.com/pkg/errors#WithMessage

Do you want to check file existence like you do in build program?

Reply:
Adding context is very useful thanks. that's for pointing out the `withMessage`.  Will update across the board with this.

Though (a) if these files aren't there then the peer's probably gone wrong and (b) the peer doesn't really report much back to if one these programs fails..

Reply:
I was thinking of a scenario where the user forgot to include metadata.json in the chaincode package.